x-pack/plugins/fleet/server/routes/security.ts

```diff
@@ -44,60 +44,41 @@ export function checkSuperuser(req: KibanaRequest) {
-      { requireLoginAction: false } // exclude login access requirement
-    );
-    return !!hasAllRequested;
-  }
-
-  return true;
+function getAuthorizationFromPrivileges(
+  kibanaPrivileges: Array<{
+    resource?: string;
+    privilege: string;
+    authorized: boolean;
+  }>,
+  searchPrivilege: string
+) {
+  const privilege = kibanaPrivileges.find((p) => p.privilege.includes(searchPrivilege));
+  return privilege ? privilege.authorized : false;
 }
 
 export async function getAuthzFromRequest(req: KibanaRequest): Promise<FleetAuthz> {
   const security = appContextService.getSecurity();
 
   if (security.authz.mode.useRbacForRequest(req)) {
-    if (checkSuperuser(req)) {
-      // Superusers get access to everything
-      // Once we implement Kibana RBAC, remove this and use `checkPrivileges` exclusively
-      return calculateAuthz({
-        fleet: { all: true, setup: true },
-        integrations: { all: true, read: true },
-        isSuperuser: true,
-      });
-    } else if (await checkFleetSetupPrivilege(req)) {
-      // fleet-setup privilege only gets access to setup actions
-      return calculateAuthz({
-        fleet: { all: false, setup: true },
-        integrations: { all: false, read: false },
-        isSuperuser: false,
-      });
-    } else {
-      // All other users only get access to read integrations if they have the read privilege
-      const checkPrivileges = security.authz.checkPrivilegesDynamicallyWithRequest(req);
-      const { privileges } = await checkPrivileges({
-        kibana: [security.authz.actions.api.get('integrations-read')],
-      });
-
-      const [intRead] = privileges.kibana;
-
-      // Once we implement Kibana RBAC, use `checkPrivileges` for all privileges instead of only integrations.read
-      return calculateAuthz({
-        fleet: { all: false, setup: false },
-        integrations: { all: false, read: intRead.authorized },
-        isSuperuser: false,
-      });
-    }
+    const checkPrivileges = security.authz.checkPrivilegesDynamicallyWithRequest(req);
+    const { privileges } = await checkPrivileges({
+      kibana: [
+        security.authz.actions.api.get('fleet-all'),
+        security.authz.actions.api.get('integrations-all'),
+        security.authz.actions.api.get('integrations-read'),
+        security.authz.actions.api.get('fleet-setup'),
+      ],
+    });
+    const fleetAllAuth = getAuthorizationFromPrivileges(privileges.kibana, 'fleet-all');
+    const intAllAuth = getAuthorizationFromPrivileges(privileges.kibana, 'integrations-all');
+    const intReadAuth = getAuthorizationFromPrivileges(privileges.kibana, 'integrations-read');
+    const fleetSetupAuth = getAuthorizationFromPrivileges(privileges.kibana, 'fleet-setup');
+
+    return calculateAuthz({
+      fleet: { all: fleetAllAuth, setup: fleetSetupAuth },
+      integrations: { all: intAllAuth, read: intReadAuth },
+      isSuperuser: false,
```

Comment:
I think we will need to change this line in order to support the endpoint that requires superuser (upload package).

```suggestion
      isSuperuser: checkSuperuser(req),
```